newIDE/app/src/InstancesEditor/InstancePropertiesEditor/index.js

```diff
@@ -21,228 +23,300 @@ import VariablesList, {
+          getChoices: () => enumerateLayers(layout),
+          getValue: (instance: gdInitialInstance) => instance.getLayer(),
+          setValue: (instance: gdInitialInstance, newValue: string) =>
+            instance.setLayer(newValue),
+        },
+        {
+          name: 'Custom size',
+          getLabel: () => i18n._(t`Custom size`),
+          valueType: 'boolean',
+          getValue: (instance: gdInitialInstance) => instance.hasCustomSize(),
+          setValue: (instance: gdInitialInstance, newValue: boolean) => {
+            if (newValue) {
+              const [width, height] = onGetInstanceSize(instance);
+              instance.setCustomWidth(width);
+              instance.setCustomHeight(height);
+            }
+            instance.setHasCustomSize(newValue);
+          },
+        },
+        {
+          name: 'custom-size-row',
+          type: 'row',
+          children: [
+            {
+              name: 'Width',
+              getLabel: () => i18n._(t`Width`),
+              valueType: 'number',
+              getValue: (instance: gdInitialInstance) =>
+                instance.getCustomWidth(),
+              setValue: (instance: gdInitialInstance, newValue: number) => {
+                instance.setHasCustomSize(true);
+                instance.setCustomWidth(Math.max(newValue, 0));
+              },
+            },
+            {
+              name: 'Height',
+              getLabel: () => i18n._(t`Height`),
+              valueType: 'number',
+              getValue: (instance: gdInitialInstance) =>
+                instance.getCustomHeight(),
+              setValue: (instance: gdInitialInstance, newValue: number) => {
+                instance.setHasCustomSize(true);
+                instance.setCustomHeight(Math.max(newValue, 0));
+              },
+            },
+          ],
         },
       ],
-    },
-  ];
-
-  _renderEmpty() {
-    return (
-      <EmptyMessage>
-        <Trans>
-          Click on an instance in the scene to display its properties
-        </Trans>
-      </EmptyMessage>
+      [i18n, layout, onEditObjectByName, onGetInstanceSize]
     );
-  }
 
-  _renderInstancesProperties() {
-    const { project, layout, instances } = this.props;
-    const instance = instances[0];
-    const associatedObjectName = instance.getObjectName();
-    const object = getObjectByName(project, layout, associatedObjectName);
-    //TODO: multiple instances support
-    const properties = instance.getCustomProperties(project, layout);
-    const instanceSchema = propertiesMapToSchema(
-      properties,
-      (instance: gdInitialInstance) =>
-        instance.getCustomProperties(project, layout),
-      (instance: gdInitialInstance, name, value) =>
-        instance.updateCustomProperty(name, value, project, layout)
+    const getSchema3D = React.useCallback(
+      () => {
+        const baseSchema = cloneDeep(schema);
+        const zOrderFieldIndex = baseSchema.findIndex(
```

Comment:
You could not do this in reorganizeSchemaFor3DInstance?

Reply:
I wanted to make a statement in the code that the Z order is not used in 3D but I can put it there I think